### ISSUE TITLE
cmd/go: Retract field missing from 'go help mod edit' definition of the GoMod struct

### ISSUE THREAD

**Issue by nomad-software** [golang/go#43281]
### What version of Go are you using (`go version`)?

<pre>
v1.16 beta1
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/tmp/go-cache"
GOENV="/home/gary/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/tmp/go-mod-cache"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/media/Data/Projects/Go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/opt/go"
GOSUMDB="sum.golang.org"
GOTMPDIR="/tmp"
GOTOOLDIR="/opt/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.16beta1"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/media/Data/Projects/Go/src/github.com/nomad-software/vend/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3728920539=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?
Issue the following command
<pre>
go help mod edit
</pre>

### What did you expect to see?
Complete information regarding the new retract flag in regard to the json struct types.

### What did you see instead?
<pre>
...
type Module struct {
		Path string
		Version string
	}

	type GoMod struct {
		Module  Module
		Go      string
		Require []Require
		Exclude []Module
		Replace []Replace
	}

	type Require struct {
		Path string
		Version string
		Indirect bool
	}

	type Replace struct {
		Old Module
		New Module
	}

	type Retract struct {
		Low       string
		High      string
		Rationale string
	}
...
</pre>
Notice that the `GoMod` definition is missing the `Retract` field.

**Comment by bcmills** [comment]
> ### What did you expect to see?
> Complete information regarding the new retract flag in regard to the json struct types.

What, specifically, did you have in mind? The JSON struct definition is already there (https://tip.golang.org/cmd/go/#hdr-Edit_go_mod_from_tools_or_scripts):
> The -retract=version and -dropretract=version flags add and drop a retraction on the given version. The version may be a single version like "v1.2.3" or a closed interval like "[v1.1.0,v1.1.9]". Note that -retract=version is a no-op if that retraction already exists.
>
> …
>
> ```go
> type Retract struct {
> 	Low       string
> 	High      string
> 	Rationale string
> }
> ```
>
> Retract entries representing a single version (not an interval) will have the "Low" and "High" fields set to the same value.

CC @jayconrod @matloob

**Comment by nomad-software** [comment]
> What, specifically, did you have in mind?

The documentation needs to add the `Retract` field to the `GoMod` struct, like this:
```
	type GoMod struct {
		Module  Module
		Go      string
		Require []Require
		Exclude []Module
		Replace []Replace
		Retract []Retract
	}
```
